.github/workflows/build.yml

```diff
@@ -0,0 +1,96 @@
+name: Build
+
+on:
+  push: 
+    branches: [staging, master]
+  pull_request:
+    branches: [staging, master]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Lint Check
+        run: yarn lint
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Lint Check
+        run: yarn test
+  
+  codecov:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Build App
+        run: yarn run build 
+      
+      - name: Upload Codecov Reports
+        uses: codecov/codecov-action@v2
```

Comment:
this probably won't do anything, codecov should be run as a step on the test job

Reply:
codecov is also probably not dependent on `yarn run build`

Reply:
> I would just run the Codecov bash script after the tests finish running.

You might have meant the codecov action, but just wanted to clarify.

The official codecov action (which Robert is currently using) is a good alternative to manually running the bash script. I'm using it already in our android/kotlin repos and have had no issues with it.

Reply:
I've only used the bash script. Obviously I haven't been following what Codecov has been doing recently as they released a new version of the uploader and are deprecating the bash uploader. It won't function after Feb 1st 2022 with intentional brownouts starting last month.
https://about.codecov.io/blog/introducing-codecovs-new-uploader/
https://docs.codecov.com/docs/about-the-codecov-bash-uploader

Sounds like we should migrate everything to the Codecov Github Action. Thanks for bringing that up :)